### PR TITLE
WithBroker(true) now fails at build time, not at runtime

### LINKED ISSUE #4768: [Bug] Non reported breaking change on 4.61.0
### Library version used

4.61.0

### .NET version

net8.0-windows10.0.17763.0

### Scenario

PublicClient - desktop app

### Is this a new or an existing app?

The app is in production, and I have upgraded to a new version of MSAL

### Issue description and reproduction steps

Upgrading to non-major versions of MSAL.Net should not have breaking changes, and 4.61.0 had a breaking change.

Upgrading to 4.61.0, we then got this reported:
https://github.com/microsoft/msstore-cli/issues/38

The release notes made no mention of this, and since it was a non-major version, this seemed safe to upgrade. We only have unit tests in our project, so we didn't get this in our builds.

### Relevant code snippets

My code was basically:
```csharp
using Microsoft.Identity.Client;
using Microsoft.Identity.Client.Extensions.Msal;
...
var app = PublicClientApplicationBuilder
    .Create(ClientId)
    .WithDefaultRedirectUri()
    .WithBroker()
    .WithParentActivityOrWindow(Helpers.NativeMethods.GetConsoleOrTerminalWindow)
    .Build();
```

That was using the Boolean overload of the WithBroker method.

### Expected behavior

No breaking change!

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

4.60.3

### Solution and workarounds

Actually upgrade to 4.61.0:
https://github.com/microsoft/msstore-cli/commit/9bf51e9b1516c3c93ce81f36164ee5eb31789e81

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #4768

We received feedback that after removing net6-windows tfm, apps fail at runtime due to using old broker API. It would be better if they fail at build time.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 1]
```diff
@@ -138,46 +138,42 @@ public PublicClientApplicationBuilder WithIosKeychainSecurityGroup(string keycha
             return this;
         }
 
+#if ANDROID || iOS
         /// <summary>
-        /// Brokers enable Single-Sign-On, device identification,
+        /// Brokers (Microsoft Authenticator, Microsoft Company Portal) enable Single-Sign-On, device identification,
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:143]:
```suggestion
        /// Brokers (Microsoft Authenticator, Intune Company Portal) enable Single-Sign-On, device identification,

```

**Review by trwalke — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 2]
```diff
@@ -138,46 +138,43 @@ public PublicClientApplicationBuilder WithIosKeychainSecurityGroup(string keycha
...
+        /// Brokers (Microsoft Authenticator, Intune Company Portal) enable Single-Sign-On, device identification,
+
         /// and application identification verification. To enable one of these features,
-        /// you need to set the WithBroker() parameters to true. See https://aka.ms/msal-net-wam 
-        /// for more information on platform specific settings required to enable the broker.
-        /// 
-        /// On iOS and Android, Authenticator and Company Portal serve as brokers.
-        /// On Windows, WAM (Windows Account Manager) serves as broker. See https://aka.ms/msal-net-wam
+        /// you need to set the WithBroker(bool) parameters to true on Android and iOS. 
+        /// On desktop platoforms, install the NuGet package Microsoft.Identity.Client.Broker and call the extension method .WithBroker(BrokerOptions
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:147]:
```suggestion
        /// On desktop platforms, install the NuGet package Microsoft.Identity.Client.Broker and call the extension method .WithBroker(BrokerOptions

```

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 3]
```diff
@@ -138,46 +138,43 @@ public PublicClientApplicationBuilder WithIosKeychainSecurityGroup(string keycha
...
-                    "\n\rInstall the NuGet package Microsoft.Identity.Client.Broker and call the extension method .WithBroker(BrokerOptions). " +
-                    "\n\rFor details, see https://aka.ms/msal-net-wam");
-            }
-#endif
             Config.IsBrokerEnabled = enableBroker;
             return this;
         }
+#else
+        /// <summary>
+        ///
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:162]: good to have a summary here

**Review by gladjohn — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:163]: The obsolete message is what matters.

**Pull request merged**